pkg/storegateway/indexheader/binary_reader.go

```diff
@@ -363,10 +363,10 @@ func (fw *FileWriter) Remove() error {
 	return os.Remove(fw.name)
 }
 
-func (w *binaryWriter) AddIndexMeta(indexVersion int, indexPostingOffsetTable uint64) error {
+func (w *binaryWriter) AddIndexMeta(indexVersion int, indexLabelOffsetTableOffset uint64) error {
 	w.buf.Reset()
 	w.buf.PutByte(byte(indexVersion))
-	w.buf.PutBE64(indexPostingOffsetTable)
+	w.buf.PutBE64(indexLabelOffsetTableOffset)
```

Comment:
The index header is stored on a persistent disk and so reused after a restart. Isn't this a breaking change, cause we assume this field to contain `indexLabelOffsetTableOffset` while it **may** contain `indexPostingOffsetTable`?

Reply:
I think this change is safe, as the posting list reader stops once it reaches the end of the posting list, even if we've fetched too much of the index file from object storage. (Each [posting list](https://github.com/prometheus/prometheus/blob/main/tsdb/docs/format/index.md#postings) starts with its length.)

Both before and after this change, we'll fetch at least as much of the index file as we need. 

However, we won't see any impact from this change unless we flush all index headers stored on disk in the store gateways after deploying this change.

Reply:
I see what you mean, but my main concern is about future uses. After this PR, we assume the meta contains `indexLabelOffsetTableOffset`, so future changes may just use it, but in reality it could be either `indexLabelOffsetTableOffset` or `indexPostingOffsetTable` depending whether the index-header on disk was created with the old or new version.

Reply:
It isn't a breaking change because the way that this value is used does not change. The only change in behaviour is when creating a new index header.

Reply:
What if we change the name we use to make the uncertainty clear to our future selves? Something like `indexSomewhereAfterEndOfLastPostingList`, but less wordy :)